Realm/Tests/TransactionTests.m

```diff
@@ -52,4 +83,102 @@ - (void)testTransactionMisuse
     XCTAssertThrows([realm deleteObject:obj], @"Outside writetransaction");
 }
 
+- (void)testDeleteBug
+{
+    RLMRealm *realm = [RLMRealm defaultRealm];
+    [realm transactionWithBlock:^{
```

Comment:
Realms are reset between all tests, so it shouldn't be necessary to delete all objects at the beginning of tests.